services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/allocator/Ipv4AddrAllocator.java

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.privateipmanager.allocator;
+
+import com.futurewei.alcor.privateipmanager.exception.IpAddrInvalidException;
+import com.futurewei.alcor.privateipmanager.exception.IpAddrNotEnoughException;
+import com.futurewei.alcor.privateipmanager.utils.Ipv4AddrUtil;
+
+import java.util.*;
+
+public class Ipv4AddrAllocator implements IpAddrAllocator {
```

Comment:
Let us add a section on the design doc about the allocation algorithm :-)

Reply:
okay，I will do that.